src/runtime/MapObject.cpp

```diff
@@ -97,8 +97,18 @@ Value MapObject::get(ExecutionState& state, const Value& key)
     return Value();
 }
 
-Value MapObject::getOrInsert(ExecutionState& state, const Value& key, const Value& value)
+Value canonicalizeKeyedCollectionKey(Value& key)
```

Comment:
Is this method commonly used (or will be) ?
If so, then it would be better to add this method as a member function of `Value`
If not, then please defining this method as `static` function because it is used only in this source file